File: dietideals-service/src/main/java/it/uninastudents/dietidealsservice/model/entity/Offerta.java
package it.uninastudents.dietidealsservice.model.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import it.uninastudents.dietidealsservice.job.TermineAstaJob;
import it.uninastudents.dietidealsservice.model.base.BaseEntity;
import it.uninastudents.dietidealsservice.model.entity.enums.StatoAsta;
import it.uninastudents.dietidealsservice.model.entity.enums.StatoOfferta;
import it.uninastudents.dietidealsservice.model.entity.enums.TipoAsta;
import it.uninastudents.dietidealsservice.service.AstaService;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.quartz.JobBuilder;
import org.quartz.JobDetail;
import org.quartz.Scheduler;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.ApplicationContext;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;

import java.math.BigDecimal;

@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@Data
@Entity
@Table(name = "offerta")
public class Offerta extends BaseEntity {

    @ManyToOne
    @JoinColumn(nullable = false)
    private Utente utente;

    @EqualsAndHashCode.Exclude
    @JsonIgnore
    @ManyToOne
    @JoinColumn(nullable = false)
    private Asta asta;

    @Column(nullable = false)
    private BigDecimal prezzo;

    @NotNull
    @Column(nullable = false)
    private StatoOfferta stato;


}

File: dietidealsapp/app/src/main/java/com/example/dietideals_app/model/Offerta.kt
package com.example.dietideals_app.model

import com.example.dietideals_app.model.enum.StatoOfferta
import java.math.BigDecimal
import java.time.OffsetDateTime
import java.util.UUID

class Offerta(
    val prezzo: BigDecimal,
    val stato: StatoOfferta,
    val utente: Utente,
    val asta: Asta,
    val creationDate: OffsetDateTime,
    val id: UUID
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Offerta) return false

        if (prezzo != other.prezzo) return false
        if (stato != other.stato) return false
        if (utente != other.utente) return false
        return asta == other.asta
    }

    override fun hashCode(): Int {
        var result = prezzo.hashCode()
        result = 31 * result + stato.hashCode()
        result = 31 * result + utente.hashCode()
        result = 31 * result + asta.hashCode()
        return result
    }
}
